src/App.tsx

```diff
@@ -1,29 +1,57 @@
-import React, { Suspense } from 'react';
+import React, { Suspense, useMemo } from 'react';
 import './App.less';
-import { ConnectionProvider } from './utils/connection';
-import { WalletProvider } from './utils/wallet';
+import { ConnectionProvider, useConnectionConfig } from './utils/connection';
+import { WalletProvider } from '@solana/wallet-adapter-react';
 import { GlobalStyle } from './global_style';
 import { Spin } from 'antd';
 import ErrorBoundary from './components/ErrorBoundary';
 import { Routes } from './routes';
 import { PreferencesProvider } from './utils/preferences';
 import { ReferrerProvider } from './utils/referrer';
+import { PhantomWalletAdapter } from '@solana/wallet-adapter-phantom';
+import { LedgerWalletAdapter } from '@solana/wallet-adapter-ledger';
+import {
+  SolletExtensionWalletAdapter,
+  SolletWalletAdapter,
+} from '@solana/wallet-adapter-sollet';
+import { SolflareWalletAdapter } from '@solana/wallet-adapter-solflare';
+import { MathWalletAdapter } from '@solana/wallet-adapter-mathwallet';
+import { WalletAdapterNetwork } from '@solana/wallet-adapter-base';
+
+function AppImpl() {
+  const { endpoint } = useConnectionConfig();
+  const network = useMemo(() => endpoint as WalletAdapterNetwork, [endpoint]);
+  const wallets = useMemo(
+    () => [
+      new SolletWalletAdapter({ network }),
+      new SolletExtensionWalletAdapter({ network }),
+      new LedgerWalletAdapter(),
+      new SolflareWalletAdapter({ network }),
+      new PhantomWalletAdapter(),
+      new MathWalletAdapter(),
+    ],
+    [network],
+  );
```

Comment:
Hi @steveluscher, really appreciate this PR! Just went through it and it looks ready to merge. Super excited for the new Wallet Adapter, but I had one question regarding this part. 

So, from what I understood from the Dialect podcast with yourself and Jordan, I was under the impression that we wouldn't have to manually import individual wallet-adapter packages, and the new Wallet Adapter would automatically recognise the wallets available. Does that sound wrong? Or is this part just for the wallets that don't support the new Wallet Adapter yet?

Reply:
That's right!

* If the user's on mobile, the Mobile Wallet Adapter will:
  * Find whatever mobile wallet app they have installed and open it, or
  * Redirect to https://solanamobile.com/wallets
* If the user's on desktop and has a **Wallet Standard** compatible wallet, it will show up in the modal automatically
* If the user's on desktop and has a wallet that conforms to the **legacy adapter interface** then it _needs_ one of these plugins to work.

The general idea is that, over time, we can start to drop these wallet adapter plugins, then eventually eliminate the need to configure them at all*.

* with the exception of ‘loadable’ wallets like Torus.

Reply:
Ah I see, thanks for the clarification!